django_cte/query.py

```diff
@@ -110,6 +117,32 @@ def generate_sql(cls, connection, query, as_sql):
         params.extend(base_params)
         return " ".join(sql), tuple(params)
 
+    @classmethod
+    def is_materialized_keyword_supported(cls, connection):
+        return any(
+            (
+                (
+                    connection.vendor == 'postgresql'
+                    and connection.pg_version >= 120000
+                ),
+                (
+                    connection.vendor == 'sqlite'
+                ),
+            )
+        )
+
+    @classmethod
+    def get_materialized_keyword(cls, connection, cte):
+        if cte.materialized:
+            if cls.is_materialized_keyword_supported(connection):
+                return 'MATERIALIZED'
+            else:
+                warnings.warn(
+                    'MATERIALIZED statement does not supported by %s. '
+                    'Parameter will be skipped by %s.'
+                    % (connection.vendor, str(cte)))
```

Comment:
I would prefer to not do this check here and instead let an error bubble up from the database if the keyword is not supported.
```suggestion
            return 'MATERIALIZED'
```

Reply:
I wanted to protect users from the ability to produce incorrect query because materialized option is like a hint, it can't change the result. But `is_materialized_keyword_supported` method is really not perfect so I removed this code.